lib/ansible/executor/task_executor.py

```diff
@@ -758,9 +758,14 @@ def _get_action_handler(self, connection, templar):
         Returns the correct action plugin to handle the requestion task action
         '''
 
+        network_group_modules = frozenset(['eos', 'nxos', 'ios', 'iosxr', 'junos', 'vyos'])
+        module_prefix = self._task.action.split('_')[0]
+
         # let action plugin override module, fallback to 'normal' action plugin otherwise
         if self._task.action in self._shared_loader_obj.action_loader:
             handler_name = self._task.action
+        elif all((module_prefix in network_group_modules, module_prefix in self._shared_loader_obj.action_loader)):
```

Comment:
module_prefix is generated by splitting on `_` so we need to be careful to not hit a false positive here